packages/devtools_app/lib/src/auto_dispose.dart

```diff
@@ -0,0 +1,72 @@
+// Copyright 2019 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:async';
+
+import 'package:meta/meta.dart';
+
+import 'ui/fake_flutter/fake_flutter.dart';
+
+/// Provides functionality to simplify listening to streams and ValueNotifiers.
+///
+/// See also:
+/// * [AutoDisposeMixin], which integrates this functionality with [State]
+///   objects.
+mixin AutoDisposeBase {
```

Comment:
this is a refactor to reuse this code in the controllers and the Flutter State classes. The handling of streams already existed. The handling of Listenable is new.